test/libsolidity/syntaxTests/structs/recursion/recursive_struct_forward_reference.sol

```diff
@@ -4,7 +4,7 @@ contract C {
     function f(Data.S memory a) public {}
 }
 contract Data {
-    struct S { S x; }
+    struct S { S[] x; }
```

Comment:
The original test would cause a different fatal error now, right?

Reply:
Yep - but that error would be triggered earlier and independently of where the struct is referenced.